eth2/utils/ssz_types/src/bitfield.rs

```diff
@@ -0,0 +1,1114 @@
+            };
+
+            Some(*byte & 1 << (i % 8) > 0)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the number of bits stored in `self`.
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    /// Returns `true` if `self.len() == 0`.
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+
+    /// Returns the underlying bytes representation of the bitfield.
+    pub fn into_raw_bytes(self) -> Vec<u8> {
+        self.bytes
+    }
+
+    /// Returns a view into the underlying bytes representation of the bitfield.
+    pub fn as_slice(&self) -> &[u8] {
+        &self.bytes
+    }
+
+    /// Instantiates from the given `bytes`, which are the same format as output from
+    /// `self.into_raw_bytes()`.
+    ///
+    /// Returns `None` if:
+    ///
+    /// - `bytes` is not the minimal required bytes to represent a bitfield of `bit_len` bits.
+    /// - `bit_len` is not a multiple of 8 and `bytes` contains set bits that are higher than, or
+    /// equal to `bit_len`.
+    fn from_raw_bytes(bytes: Vec<u8>, bit_len: usize) -> Option<Self> {
+        if bit_len == 0 {
+            if bytes.len() == 1 && bytes == [0] {
+                // A bitfield with `bit_len` 0 can only be represented by a single zero byte.
+                Some(Self {
+                    bytes,
+                    len: 0,
+                    _phantom: PhantomData,
+                })
+            } else {
+                None
+            }
+        } else if bytes.len() != bytes_for_bit_len(bit_len) || bytes.is_empty() {
+            // The number of bytes must be the minimum required to represent `bit_len`.
+            None
+        } else {
+            // Ensure there are no bits higher than `bit_len` that are set to true.
+            let (mask, _) = u8::max_value().overflowing_shr(8 - (bit_len as u32 % 8));
+
+            if (bytes.first().expect("Guarded against empty bytes") & !mask) == 0 {
+                Some(Self {
+                    bytes,
+                    len: bit_len,
+                    _phantom: PhantomData,
+                })
+            } else {
+                None
+            }
+        }
+    }
+
+    /// Returns the `Some(i)` where `i` is the highest index with a set bit. Returns `None` if
+    /// there are no set bits.
+    pub fn highest_set_bit(&self) -> Option<usize> {
+        let byte_i = self.bytes.iter().position(|byte| *byte > 0)?;
+        let bit_i = 7 - self.bytes[byte_i].leading_zeros() as usize;
+
+        Some((self.bytes.len().saturating_sub(1) - byte_i) * 8 + bit_i)
+    }
+
+    /// Returns an iterator across bitfield `bool` values, starting at the lowest index.
+    pub fn iter(&self) -> BitIter<'_, T> {
+        BitIter {
+            bitfield: self,
+            i: 0,
+        }
+    }
+
+    /// Returns true if no bits are set.
+    pub fn is_zero(&self) -> bool {
+        !self.bytes.iter().any(|byte| (*byte & u8::max_value()) > 0)
```

Comment:
Seems like a strange way to write this, the `& u8::max_value()` is a no-op, and `!any` is the same as `all!`:

```rust
self.bytes.iter().all(|byte| *byte == 0)
```